components/List/CartListItem.js

```diff
@@ -19,6 +19,9 @@ export const CartListItem = ({
     isError: false,
     message: "",
   });
+  useEffect(() => {
```

Comment:
not sure I follow all the code here, but this `useEffect` seems redundant. Why not pass `quantity` to the `value` and `quantityHandlerChange` to `onChange` directly  on the input? seems like it's the parent of this component that should control the state.